tools/rosout/rosout.cpp

```diff
@@ -82,10 +83,13 @@ class Rosout
 
   void init()
   {
-    bool disable_file_logging = false;
-    node_.getParamCached("/rosout/disable_file_logging", disable_file_logging);
-
-    if (!disable_file_logging)
+    const char* disable_file_logging = getenv("ROSOUT_DISABLE_FILE_LOGGING");
+    if (!disable_file_logging  // Not set.
+      || !disable_file_logging[0]  // Set to empty string.
+      || strcmp(disable_file_logging, "0") == 0
+      || boost::iequals(disable_file_logging, "false")
+      || boost::iequals(disable_file_logging, "off")
+      || boost::iequals(disable_file_logging, "no"))
```

Comment:
I am just wondering if the case insensitive comparison if worth the boost dependency? Each condition could be duplicates to check for lower case as well as upper case (not allowing mixed case)...

Reply:
You don't need boost for case-insensitivity.  Just convert the output to lower case:

https://github.com/ros/ros_comm/pull/1425#issuecomment-403805515